cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleBuilder.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule;
+
+import co.cask.cdap.api.schedule.ConstraintProgramScheduleBuilder;
+import co.cask.cdap.api.schedule.ScheduleBuilder;
+import co.cask.cdap.internal.app.runtime.schedule.trigger.PartitionTrigger;
+import co.cask.cdap.internal.app.runtime.schedule.trigger.TimeTrigger;
+import co.cask.cdap.internal.schedule.ScheduleCreationSpec;
+import co.cask.cdap.proto.ProtoConstraint;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * The default implementation of {@link ScheduleBuilder}.
+ */
+public class DefaultScheduleBuilder implements ConstraintProgramScheduleBuilder {
+
+  private final String name;
+  private final NamespaceId namespace;
+  private final String programName;
```

Comment:
I don't want to put `ProgramId` into the builder, because we don't know the appName.
When building, the user can still call `setName` on the app configurer afterwards.